lib/monitor.toit

```diff
@@ -165,6 +165,60 @@ monitor Signal:
       await: current_ >= awaited
       if condition.call: return
 
+/**
+A synchronization gate.
+
+The gate can be open or closed. When a task tries to $enter, it waits
+  until the gate is open.
+*/
+class Gate:
+  signal_ /Signal ::= Signal
+  open_ /bool := ?
+
+  /**
+  Constructs a new gate.
+
+  If $open is true, starts with the gate open.
+  */
+  constructor --open/bool=false:
+    open_ = open
+
+  /**
+  Opens the gate.
+
+  Does nothing if the gate is already open.
+  */
+  open -> none:
+    if is_open: return
+    open_ = true
+    signal_.raise
+
+  /**
+  Closes the gate.
+
+  Any task that is trying to $enter will block until the gate is opened again.
+  */
+  close:
```

Comment:
I'm not super excited about this being called 'close'. It is very different from other uses of close.

Maybe we can use lock/unlock instead?

Reply:
done.